app/models/automate_workspace.rb

```diff
@@ -18,4 +18,33 @@ def merge_output!(hash)
     save!
     self
   end
+
+  def decrypt(object_name, attribute)
+    MiqPassword.decrypt(encrypted_value(object_name, attribute))
```

Comment:
We have to be careful this doesn't expose us to an [Oracle attack](https://en.wikipedia.org/wiki/Oracle_attack).  For example, a user can potentially manually create an entry with the string "password:v2{23423h42h}".  If we aren't careful with the error message, they can use that information to determine "good" v2 keys from "bad" v2 keys.

cc @abellotti @gtanzillo

Reply:
@Fryguy 
The caller never provides the encrypted value. Instead inside of an Automate Workspace identified with a GUID, they pass in the object name and attribute name. 
When the call comes into the server we locate the Automate Workspace and use the object nam e and attribute name to get the encrypted value.

For this to be an oracle attack they would have to guess the GUID of the Automate Workspace, then look at objects and the attribute name and be able to tweak the encrypted value and then call this API to get the decrypted value.